plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -681,17 +681,34 @@ class AzBatchService implements Closeable {
                 .withContainerConfiguration(containerConfig)
     }
 
-    protected StartTask createStartTask() {
-        if( config.batch().getCopyToolInstallMode() != CopyToolInstallMode.node )
-            return null
 
+    protected StartTask createStartTask(AzPoolOpts opts) {
+
+        def startCmd = []
         final resourceFiles = new ArrayList(10)
-        resourceFiles << new ResourceFile()
-            .withHttpUrl(AZCOPY_URL)
-            .withFilePath('azcopy')
+
+
+        // Get any custom start task command
+        log.debug "Adding custom start task to command: ${opts.startTask}"
+        if ( opts.startTask ) {
+            log.debug "Adding custom start task to command: ${opts.startTask}"
+            startCmd << opts.startTask
+        }
+
+
+        // If enabled, append azcopy installer to start task command
+        if( config.batch().getCopyToolInstallMode() == CopyToolInstallMode.node )
+            startCmd << 'chmod +x azcopy && mkdir \$AZ_BATCH_NODE_SHARED_DIR/bin/ && cp azcopy \$AZ_BATCH_NODE_SHARED_DIR/bin/'
+            resourceFiles << new ResourceFile()
+                .withHttpUrl(AZCOPY_URL)
+                .withFilePath('azcopy')
+            log.debug "Adding azcopy installer to start task"
+
+        final startTaskCmd = "bash -c \"${startCmd.join('; ')}\""
```

Comment:
What if thre's no startTask script and copyToolInstallMode != node? this is going to be empty.

Reply:
It would be nice to add a unit test to check different conditions.